### PR TITLE
[Merged by Bors] - p2p node: Package improvements

### PR DESCRIPTION
**Pull request by josebalius**
## Motivation
As stated in Discord, I have been reviewing the whole project and hope to open a series of PRs that clean up a few things in the codebase and make things more idiomatic in general. There's a ton of good Go code here and my hope is that these changes can only make it better, more dependable, and idiomatic.

Instead of opening a huge PR that spans several packages, I figured it would be best to open smaller PRs that are package-focused. I have started with the p2p package and have several PRs to come.

## Changes
- Remove unneeded pkg var: `emptyNode`
- Clean up resource leak potentials with open file handlers
- Add more context to errors
- Idiomatic variable naming

## Test Plan
Unit tests all pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: p2p/node/helpers.go
Comment (josebalius) [inline p2p/node/helpers.go:37]: Re-use named return which stops the code from needing an "emptyNode" variable in the package.

File: p2p/node/localnode.go
Comment (josebalius) [inline p2p/node/localnode.go:46]: Add context to these errors, so that we can tell when a failure happens from parsing each type of key.

File: p2p/node/localnodestore.go
Comment (josebalius) [inline p2p/node/localnodestore.go:43]: `ioutil.WriteFile` is a much better encapsulation for what we are trying to do here. The `0666` file mode is also the same that `File.Write` would usually do.

Also, the previous implementation had a potential resource leak by leaving a file handle open if either `f.Write` or `f.Sync` failed. File handle close defers should happen immediately after the file is created so that we can clean up in all cases.

File: p2p/node/localnodestore.go
Comment (josebalius) [inline p2p/node/localnodestore.go:46]: This log call feels like it's best done by callers or at the very least we should accept a `logger`

File: p2p/node/localnodestore.go
Comment (josebalius) [inline p2p/node/localnodestore.go:69]: The previous implementation had a potential resource leak. If the `io.Copy` failed, we would fail to close the file handle. `io.ReadFile` can take care of the whole operation for us and we fix two birds with one stone sort of speak.

File: p2p/node/localnodestore.go
Comment (josebalius) [inline p2p/node/localnodestore.go]: The doc here might need updating, I don't see how this function is reading a "random" node from the data folder. We seem to fetch them all and read the first one. Does it need updating?

Comment (y0sher) [reply p2p/node/localnodestore.go]: yes you are correct, it is documenting older behavior..

**Review by y0sher — APPROVED** [review]

**Comment by y0sher** [comment]
Thanks for doing this! I appreciate the thought and work.

File: p2p/node/localnodestore.go
Comment (josebalius) [reply p2p/node/localnodestore.go]: Thanks, I'll update the comment and remove the TODO.

**Comment by josebalius** [comment]
@y0sher looks like I need a maintainer's approval to have CI run. Do you know if they will automatically take a look or should I ping someone?

**Comment by y0sher** [comment]
Thanks for reminding, approved for running the CI

**Comment by nkryuchkov** [comment]
bors try

**Review by nkryuchkov — APPROVED** [review]
Good job! Thanks for your PR

**Comment by nkryuchkov** [comment]
bors merge

**Pull request closed without merging**
